nightingale_body/src/nightingale_body/body_control.py

```diff
@@ -46,17 +50,55 @@ def _make_goal(self, jnt_name, jnt_pos, ee_vel=0.5, ee_accel=0.5, timeout=5):
         return goal
 
     def cmd_head(self, target_jnt_name, target_jnt_pos):
+        if self.head_is_close(target_jnt_name, target_jnt_pos):
+            return True
+
         goal = self._make_goal(target_jnt_name, target_jnt_pos)
 
         self.head_ac.send_goal(goal)
         return self.head_ac.wait_for_result()
 
     def cmd_torso(self, target_jnt_name, target_jnt_pos):
+        if self.torso_is_close(target_jnt_name, target_jnt_pos):
+            return True
+
         goal = self._make_goal(target_jnt_name, target_jnt_pos)
 
         self.torso_ac.send_goal(goal)
         return self.torso_ac.wait_for_result()
 
+    def head_is_close(self, target_jnt_name, target_jnt_pos):
+        curr_jnt_state = rospy.wait_for_message("/movo/head/joint_states", JointState)
+        curr_jnt_pos = {
+            name: pos for name, pos in zip(curr_jnt_state.name, curr_jnt_state.pos)
+        }
+
+        diff = np.zeros(len(target_jnt_name))
+        for idx, (tgt_name, tgt_pos) in enumerate(zip(target_jnt_name, target_jnt_pos)):
+            try:
+                diff[idx] = np.abs(curr_jnt_pos[tgt_name] - tgt_pos)
+            except KeyError:
+                continue
+
+        return np.sum(diff) < self.CLOSE_ANGLE_TOL
+
+    def torso_is_close(self, target_jnt_name, target_jnt_pos):
+        curr_jnt_state = rospy.wait_for_message(
+            "/movo/linear_actuator/joint_states", JointState
+        )
+        curr_jnt_pos = {
+            name: pos for name, pos in zip(curr_jnt_state.name, curr_jnt_state.pos)
+        }
+
+        diff = np.zeros(len(target_jnt_name))
+        for idx, (tgt_name, tgt_pos) in enumerate(zip(target_jnt_name, target_jnt_pos)):
```

Comment:
Isn't there only a single joint here? Is this loop necessary for the torso?

Reply:
If we are being consistent, the joint target passed in will be a list (with 1 value) in which case the loop is the easiest solution even though it only gets looped through once